### PR TITLE
Add delete button to object detail modal

### PR DESCRIPTION
**Pull request by akiselev**
Attempt at #22547 (the table grid view is complicated and adding the on-hover button will take more time)

### To Verify

1. Run Metabase with `experimental-enable-actions` flag or patch [this selector](https://github.com/metabase/metabase/blob/84bd09ffc2bc5069ff7482fb6c6b9ae70b737dd3/frontend/src/metabase/writeback/selectors.ts#L5) to return `true`
2. Click `Enable actions` in Database settings (Danger Zone!) for an H2 or Postgres database
3. Open an object detail for any row (click the PK or hover the row and click the "expand" icon on the left)
4. Click the "trash" icon next to the previous/next/edit buttons in the header
5. You should see a confirmation dialog
6. When you confirm, it should fire off an api request to `/api/actions/row/delete` 

image

image

### CODE REVIEW

File: frontend/src/metabase/writeback/api.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { POST } from "metabase/lib/api";
+
+export const deleteRow = POST("/api/actions/row/delete");
```
Comment (kulyk) [inline frontend/src/metabase/writeback/api.ts:3]: I guess the preferred way to add new endpoints on FE is to create an API object in `metabase/services`, [see an example](https://github.com/metabase/metabase/blob/9ebcf4b185b91f1031ce40865e460f32350f40ac/frontend/src/metabase/services.js#L138)

File: frontend/src/metabase/writeback/actions.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      throw new Error("Cannot delete row from table without a primary key");
+    }
+
+    const pk = field.isNumeric() && typeof id === "string" ? parseInt(id) : id;
+    const result = await api.deleteRow({
+      type: "query",
+      database: table.db_id,
+      query: {
+        "source-table": table.id,
+        filter: ["=", ["field", field.id, null], pk],
```
Comment (kulyk) [inline frontend/src/metabase/writeback/actions.ts:27]: nit: it should be possible to do `filter: ["=", field.reference(), pk]`

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx [hunk 3]
```diff
@@ -248,6 +258,15 @@ export function ObjectDetailFn({
...
 
+  let deleteRow;
+  if (canEdit) {
+    deleteRow = () =>
+      deleteRowFromObjectDetail({
+        id: zoomedRowID,
+        table,
+      });
+  }
+
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx:269]: Can we maybe just wrap it into `useCallback` or `useMemo` to make it more declerative? For instance:

```jsx
const handleDeleteRow = useCallback(() => {
   if (canEdit) {
      deleteRowFromObjectDetail({ ... })
   }
}, [ ... ])

// OR

const onDeleteRow = useMemo(() => {
   if (!canEdit) {
      return
   }
   deleteRowFromObjectDetail({ ... })
}, [...])
```

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx [hunk 4]
```diff
@@ -338,14 +361,38 @@ export function ObjectDetailHeader({
...
+            <Button
+              className="mr1"
+              icon={isEditing ? "eye" : "pencil"}
+              onClick={onToggleEditingModeClick}
+              iconSize={20}
+              onlyIcon
+              borderless
+            />
+          </>
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx:395]: Can we move the modal to its own file? There's ongoing work on this component, don't want our prototyping work cause too much noise for other folks

File: frontend/src/metabase/writeback/api.ts (hunk 1 above)
Comment (akiselev) [reply frontend/src/metabase/writeback/api.ts]: Yeah that was my first instinct but I figured we'd want to separate out the endpoints so no one else calls them accidentally.

moved to `metabase/services`

File: frontend/src/metabase/writeback/actions.ts (hunk 2 above)
Comment (akiselev) [reply frontend/src/metabase/writeback/actions.ts]: Thanks!

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx (hunk 3 above)
Comment (akiselev) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx]: Done

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx (hunk 4 above)
Comment (akiselev) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx]: Move to `metabase/writeback/components/ObjectDetails/ActionHeader`

**Review by kulyk — APPROVED** [review]
Nice! Thanks for addressing the comments

**Review by camsaul — APPROVED** [review]
It looks good to me and the request body looks right as far as I can tell.

**Pull request merged**